File: backend/src/main/java/ftn/knowledge/engineering/ComputerRecommender/model/Desktop.java
package ftn.knowledge.engineering.ComputerRecommender.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Desktop extends Base {
    private Motherboard motherboard;
    private GPU gpu;
    private PowerSupply powerSupply;
    private Storage storage;
    public Desktop(String name, String manufacturer, double priceEur, PlatformCompatibility platformCompatibility,Motherboard motherboard, GPU gpu, PowerSupply powerSupply, Storage storage){
        super(name,manufacturer,priceEur,platformCompatibility);
        this.motherboard = motherboard;
        this.gpu = gpu;
        this.powerSupply = powerSupply;
        this.storage = storage;

    }
}

File: frontend/src/features/ontology/types/Desktop.ts
interface Base {
  name: string;
  manufacturer: string;
  priceEur: string;
}

export interface Desktop extends Base {
  motherboard: Motherboard;
  gpu: Gpu;
  powerSupply: PowerSupply;
  storage: Storage;
}

export interface Motherboard extends Base {
  ram: string;
  numberOfRAMSlots: number;
  type: string;
  cpu: Cpu;
  chipset: Chipset;
}

export interface Gpu extends Base {
  GPUType: string;
  clockSpeed: number;
  boostClockSpeed: number;
  vRAMSize: number;
}

export interface Chipset extends Base {
  type: string;
}

export interface Cpu extends Base {
  generaton: string;
  clockSpeed: number;
  coreCount: number;
  threadCount: number;
  tdp: number;
  cacheSizes: number[];
  supportsMultithreading: boolean;
  supportsOverclocking: boolean;
  hasIntegratedGraphics: boolean;
}

export interface PowerSupply extends Base {
  wattage: number;
}

export interface Storage extends Base {
  type: string;
  memoryCapacity: number;
  writeSpeed: number;
}
